src/oslc/oslcmain.cpp

```diff
@@ -188,12 +188,12 @@ main (int argc, const char *argv[])
             ok = compiler.compile_buffer (sourcecode, osobuffer, args, "",
                                           shader_path);
         if (ok) {
-            std::ofstream file;
+            OIIO::ofstream file;
             OIIO::Filesystem::open (file, compiler.output_filename());
-            if (file.good()) {
+            if (file)
                 file << osobuffer;
-                file.close ();
-            }
+            else
```

Comment:
`ok = file.good()` to catch write errors in addition to open errors?

Reply:
Yes, good idea, will fix.

Reply:
updated